src/server/pfs/fuse/file.go

```diff
@@ -38,18 +83,19 @@ func (f *file) InnerFile() nodefs.File {
 }
 
 func (f *file) Read(dest []byte, offset int64) (fuse.ReadResult, fuse.Status) {
-	repo, file, err := f.fs.parsePath(f.name)
-	if err != nil {
-		return nil, toStatus(err)
+	waitn := offset + int64(len(dest))
+	if waitn > int64(f.attr.Size) {
+		waitn = int64(f.attr.Size)
 	}
-	switch {
-	case repo != nil:
-		return nil, fuse.Status(syscall.EISDIR)
-	case file != nil:
-		return newReadResult(f.fs.c, file, offset, len(dest)), fuse.OK
-	default:
-		return nil, fuse.Status(syscall.EISDIR)
+	f.counter.wait(waitn)
+	// check if there was an error reading the file
+	if f.err != nil {
```

Comment:
So ... if the file was created w a call to `newFile()` but FUSE never receives any reads (presumably calls to this `Read` function), then we never report the error? Does it make sense to have a logger emit this error right when it happens (after the `GetFile`?)

Reply:
Fair enough. I can see how it'd generate noise before a user actually attempts a read.

Reply:
Yeah, that's my main worry. The FUSE mount is often backgrounded within a shell, so if it gets too chatty it makes the shell unusable.